### ISSUE TITLE
[CRASH] UIKit __120-[UIView(UIConstraintBasedLayout) _layoutEngine_didAddLayoutConstraint:roundingAdjustm

### ISSUE THREAD

**Issue by rafaelks** [RocketChat/Rocket.Chat.iOS#814]
# Information

**Introduced in:** 1.2.1 (83)
**OS:** iOS 11
**Device:** iPad
**Occurrences so far (from 90 days ago):** 2

# Stacktrace

```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0x18376fd38 __exceptionPreprocess
1  libobjc.A.dylib                0x182c84528 objc_exception_throw
2  CoreFoundation                 0x18376fc0c +[NSException raise:format:]
3  Foundation                     0x1840fec24 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:]
4  UIKit                          0x18d663a70 __120-[UIView(UIConstraintBasedLayout) _layoutEngine_didAddLayoutConstraint:roundingAdjustment:mutuallyExclusiveConstraints:]_block_invoke_2
5  UIKit                          0x18cc68350 -[UIView(UIConstraintBasedLayout) _layoutEngine_didAddLayoutConstraint:roundingAdjustment:mutuallyExclusiveConstraints:]
6  UIKit                          0x18cc68168 -[UIView(UIConstraintBasedLayout) _tryToAddConstraintWithoutUpdatingConstraintsArray:roundingAdjustment:mutuallyExclusiveConstraints:]
7  UIKit                          0x18cc71264 __50-[UIView(UIConstraintBasedLayout) addConstraints:]_block_invoke
8  Foundation                     0x1840a059c -[NSISEngine withBehaviors:performModifications:]
9  UIKit                          0x18cc7112c -[UIView(UIConstraintBasedLayout) addConstraints:]
10 UIKit                          0x18d7205bc -[UIInputWindowController updateViewConstraints]
11 UIKit                          0x18d722cc8 -[UIInputWindowController updateForKeyplaneChangeWithContext:]
12 UIKit                          0x18d43a2a4 -[_UIRemoteKeyboards performOnControllers:]
13 UIKit                          0x18d71b914 -[UIInputSetHostView _didChangeKeyplaneWithContext:]
14 UIKit                          0x18d53001c -[_UIKBCompatInputView _didChangeKeyplaneWithContext:]
15 UIKit                          0x18cff4bb0 -[UIKeyboard _didChangeKeyplaneWithContext:]
16 UIKit                          0x18cfdbef0 -[UIKeyboardImpl _didChangeKeyplaneWithContext:]
17 UIKit                          0x18d1d9128 -[UIKeyboardLayoutStar(UIKeyboardLayoutJapanese50OnFlick) _didChangeKeyplaneWithContext:]
18 UIKit                          0x18cc76cf4 -[UIKeyboardLayoutStar setKeyplaneName:]
19 UIKit                          0x18cc753dc -[UIKeyboardLayoutStar showKeyboardWithInputTraits:screenTraits:splitTraits:]
20 UIKit                          0x18cc75004 -[UIKeyboardImpl finishLayoutChangeWithArguments:]
21 UIKit                          0x18cba633c -[UIKeyboardImpl updateLayout]
22 UIKit                          0x18cfd6ef4 -[UIKeyboardImpl updateLayoutIfNecessary]
23 UIKit                          0x18cba5e38 -[UIKeyboardImpl setFrame:]
24 UIKit                          0x18cc87548 -[UIKeyboard activate]
25 UIKit                          0x18cc872a0 -[UIKeyboardAutomatic activate]
26 UIKit                          0x18cff27c4 -[UIKeyboard setBounds:]
27 UIKit                          0x18ce5a940 -[UIView(Geometry) _applyISEngineLayoutValuesToBoundsOnly:]
28 UIKit                          0x18cb96218 -[UIView(Geometry) _resizeWithOldSuperviewSize:]
29 CoreFoundation                 0x183649bec -[__NSArrayM enumerateObjectsWithOptions:usingBlock:]
30 UIKit                          0x18cb847e8 -[UIView(Geometry) resizeSubviewsWithOldSize:]
31 UIKit                          0x18cc6dac0 -[UIView(AdditionalLayoutSupport) _is_layout]
32 UIKit                          0x18ce5dcf4 -[UIView(Hierarchy) _updateConstraintsAsNecessaryAndApplyLayoutFromEngine]
33 UIKit                          0x18cb7c000 -[UIView(CALayerDelegate) layoutSublayersOfLayer:]
34 QuartzCore                     0x18774d0b4 -[CALayer layoutSublayers]
35 QuartzCore                     0x187751194 CA::Layer::layout_if_needed(CA::Transaction*)
36 UIKit                          0x18cb90cf8 -[UIView(Hierarchy) layoutBelowIfNeeded]
37 UIKit                          0x18d71ba14 -[UIInputSetHostView layoutIfNeeded]
38 UIKit                          0x18d726f58 -[UIInputWindowController _updateBackdropViews]
39 UIKit                          0x18d72b38c __43-[UIInputWindowController setInputViewSet:]_block_invoke_2.1516
40 UIKit                          0x18d723678 -[UIInputWindowController moveFromPlacement:toPlacement:starting:completion:]
41 UIKit                          0x18d72b2cc __43-[UIInputWindowController setInputViewSet:]_block_invoke.1515
42 UIKit                          0x18d71d140 -[UIInputWindowController performOperations:withTemplateNotificationInfo:]
43 UIKit                          0x18d72ae18 -[UIInputWindowController setInputViewSet:]
44 UIKit                          0x18d72304c -[UIInputWindowController performOperations:withAnimationStyle:]
45 UIKit                          0x18cc33f24 -[UIPeripheralHost(UIKitInternal) setInputViews:animationStyle:]
46 UIKit                          0x18cba46e0 -[UIPeripheralHost(UIKitInternal) _reloadInputViewsForResponder:]
47 UIKit                          0x18d43878c -[_UIRemoteKeyboards reloadForSnapshotting]
48 UIKit                          0x18d13df78 -[UITextEffectsWindow _didSnapshot]
49 UIKit                          0x18cdeffa0 __65-[UIApplication _beginSnapshotSessionForScene:withSnapshotBlock:]_block_invoke.2257
50 UIKit                          0x18cdf0888 __65-[UIApplication _performSnapshotsWithAction:forScene:completion:]_block_invoke.2282
51 FrontBoardServices             0x185e434a0 -[FBSSceneSnapshotAction _finishAllRequests]
52 FrontBoardServices             0x185e43214 -[FBSSceneSnapshotAction executeRequestsWithHandler:completionHandler:expirationHandler:]
53 UIKit                          0x18cdf0668 __65-[UIApplication _performSnapshotsWithAction:forScene:completion:]_block_invoke
54 UIKit                          0x18cdefbf0 -[UIApplication _beginSnapshotSessionForScene:withSnapshotBlock:]
55 UIKit                          0x18cdf03a4 -[UIApplication _performSnapshotsWithAction:forScene:completion:]
56 UIKit                          0x18cdf024c -[UIApplication _handleSnapshotAction:forScene:completion:]
57 UIKit                          0x18cdebbfc __71-[UIApplication _handleSnapshotAction:forScene:deactivationCompletion:]_block_invoke
58 UIKit                          0x18cdeba08 -[UIApplication _handleSnapshotAction:forScene:deactivationCompletion:]
59 UIKit                          0x18d075c9c __98-[__UICanvasLifecycleMonitor_Compatability deactivateEventsOnly:withContext:forceExit:completion:]_block_invoke.271
60 UIKit                          0x18cdef680 _runAfterCACommitDeferredBlocks
61 UIKit                          0x18cde2b24 _cleanUpAfterCAFlushAndRunDeferredBlocks
62 UIKit                          0x18cb71808 _afterCACommitHandler
63 CoreFoundation                 0x1837178b8 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__
64 CoreFoundation                 0x183715270 __CFRunLoopDoObservers
65 CoreFoundation                 0x18371582c __CFRunLoopRun
66 CoreFoundation                 0x1836362d8 CFRunLoopRunSpecific
67 GraphicsServices               0x1854c7f84 GSEventRunModal
68 UIKit                          0x18cbe2880 UIApplicationMain
69 Rocket.Chat                    0x10043da18 main (ServerManager.swift:14)
70 libdyld.dylib                  0x18315a56c start
```

**Comment by cardoso** [comment]
I think this might have been introduced by #814 and fixed by #1016 

Let's wait for new reports from fabric 🙏

**Comment by rafaelks** [comment]
I agree! Let's close the issue on Fabric and close this issue here. We can reopen if it ressurges.
